src/Appwrite/Platform/Workers/StatsUsage.php

```diff
@@ -246,8 +323,126 @@ private function reduce(Document $project, Document $document, array &$metrics,
+                            '$id' => $project->getId(),
+                            '$internalId' => $project->getInternalId(),
+                            'database' => $project->getAttribute('database'),
+                        ]);
+                        $this->projects[$project->getInternalId()]['stats'][] = $document;
+
+                        $this->prepareForLogsDB($project, $document);
+                    }
+                }
+            } catch (Exception $e) {
+                console::error('[' . DateTime::now() . '] project [' . $project->getInternalId() . '] database [' . $project['database'] . '] ' . ' ' . $e->getMessage());
+            }
+        }
+
+        foreach ($this->projects as $internalId => $projectStats) {
+            if (empty($internalId)) {
+                continue;
+            }
+            try {
+                /** @var \Utopia\Database\Database $dbForProject */
+                $dbForProject = $getProjectDB($projectStats['project']);
+                Console::log('Processing batch with ' . count($projectStats['stats']) . ' stats');
+                $dbForProject->createOrUpdateDocumentsWithIncrease('stats', 'value', $projectStats['stats']);
+                Console::success('Batch successfully written to DB');
+
+                unset($this->projects[$internalId]);
+            } catch (Throwable $e) {
+                Console::error('Error processing stats: ' . $e->getMessage());
+            }
+        }
+
+        $this->writeToLogsDB();
+
+    }
+
+    protected function prepareForLogsDB(Document $project, Document $stat)
+    {
+        if (System::getEnv('_APP_STATS_USAGE_DUAL_WRITING', 'disabled') === 'disabled') {
+            return;
+        }
+        if (array_key_exists($stat->getAttribute('metric'), $this->skipBaseMetrics)) {
+            return;
+        }
+        foreach ($this->skipParentIdMetrics as $skipMetric) {
+            if (str_ends_with($stat->getAttribute('metric'), $skipMetric)) {
+                return;
+            }
+        }
+        $documentClone = new Document($stat->getArrayCopy());
+        $documentClone->setAttribute('$tenant', (int) $project->getInternalId());
+        $this->statDocuments[] = $documentClone;
+    }
+
+    protected function writeToLogsDB(): void
+    {
+        if (System::getEnv('_APP_STATS_USAGE_DUAL_WRITING', 'disabled') === 'disabled') {
+            Console::log('Dual Writing is disabled. Skipping...');
+            return;
+        }
+
+        /** @var \Utopia\Database\Database $dbForLogs*/
+        $dbForLogs = call_user_func($this->getLogsDB);
+        $dbForLogs
+            ->setTenant(null)
+            ->setTenantPerDocument(true);
+
+        try {
+            Console::log('Processing batch with ' . count($this->statDocuments) . ' stats');
+            $dbForLogs->createOrUpdateDocumentsWithIncrease(
+                'stats',
+                'value',
+                $this->statDocuments
+            );
```

Comment:
Have you tested locally? No issues with upsert + tenant per document?

Reply:
basic manual test looks okay